controllers/managedocs_controller_test.go

```diff
@@ -669,6 +683,65 @@ var _ = Describe("ManagedOCS controller", func() {
 				utils.WaitForResource(k8sClient, ctx, promTemplate.DeepCopy(), timeout, interval)
 			})
 		})
+		When("there is no pagerduty secret in the cluster", func() {
+			It("should not create alertmanager config secret", func() {
+				// Verify that a pagerduty secret is not present
+				secret := pdSecretTemplate.DeepCopy()
+				Expect(k8sClient.Get(ctx, utils.GetResourceKey(secret), secret)).Should(
+					WithTransform(errors.IsNotFound, BeTrue()),
+				)
+
+				// Ensure, over a period of time, that the resources are not created
+				resList := []runtime.Object{
+					amConfigSecretTemplate.DeepCopy(),
+				}
+				utils.EnsureNoResources(k8sClient, ctx, resList, timeout, interval)
+			})
+		})
+		When("there is no key for pagerduty secret in the cluster", func() {
+			It("should not create alertmanager config secret", func() {
+				// Create empty pagerduty secret
+				secret := pdSecretTemplate.DeepCopy()
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+				// Ensure, over a period of time, that the resources are not created
+				resList := []runtime.Object{
+					amConfigSecretTemplate.DeepCopy(),
+				}
+				utils.EnsureNoResources(k8sClient, ctx, resList, timeout, interval)
+
+				// Remove the secret for future cases
+				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
+			})
+		})
+		When("there is an invalid key for pagerduty secret in the cluster", func() {
+			It("should not create alertmanager config secret", func() {
+				// Create pagerduty secret with invalid key
+				secret := pdSecretTemplate.DeepCopy()
+				secret.Data["blah-blah"] = []byte("test-key")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+				// Ensure, over a period of time, that the resources are not created
+				resList := []runtime.Object{
+					amConfigSecretTemplate.DeepCopy(),
+				}
+				utils.EnsureNoResources(k8sClient, ctx, resList, timeout, interval)
+
+				// Remove the secret for future cases
+				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
+			})
+		})
```

Comment:
Why do we care about other keys, I see a case for an invalid value (which we cannot test) but an invalid key is not a case.